prov/util/src/util_coll.c

```diff
@@ -557,77 +558,113 @@ static int util_coll_proc_reduce_item(struct util_coll_mc *coll_mc,
 	return FI_SUCCESS;
 }
 
-static int util_coll_process_work_items(struct util_coll_mc *coll_mc)
+int ofi_coll_process_pending(struct util_ep *ep)
 {
+	struct util_coll_mc *coll_mc;
+	struct slist_entry *entry;
 	struct util_coll_hdr *hdr;
 	struct util_coll_xfer_item *xfer_item;
+	struct iovec iov;
+	struct fi_msg_tagged msg = { 0 };
```

Comment:
please don't over-initialize data structures.  Most of these fields will be set further down.